src/main/kotlin/org/acejump/session/Session.kt

```diff
@@ -123,34 +124,51 @@ class Session(private val editor: Editor) {
 
     when (val result = tagger.markOrJump(query, results.clone())) {
       is TaggingResult.Jump -> {
-        tagJumper.jump(result.offset, shiftMode)
-        tagCanvas.removeMarkers()
+        tagJumper.jump(result.tag, shiftMode, isCrossEditor = mainEditor !== result.tag.editor)
+        tagCanvases.values.forEach(TagCanvas::removeMarkers)
         end(result)
       }
 
       is TaggingResult.Mark -> {
-        val tags = result.tags
-        tagCanvas.setMarkers(tags)
-
-        val cache = EditorOffsetCache.new()
-        val boundaries = VISIBLE_ON_SCREEN
-
-        if (tags.none {
-            boundaries.isOffsetInside(editor, it.offsetL, cache) ||
-              boundaries.isOffsetInside(editor, it.offsetR, cache)
-          }) tagVisitor?.scrollToClosest()
+        val markers = result.markers
+        
+        for ((editor, canvas) in tagCanvases) {
+          canvas.setMarkers(markers[editor].orEmpty())
+        }
+        
+        if (jumpEditors.all { editor ->
+            val cache = EditorOffsetCache.new()
+            markers[editor].let { it == null || it.none { marker ->
+              VISIBLE_ON_SCREEN.isOffsetInside(editor, marker.offsetL, cache) ||
+              VISIBLE_ON_SCREEN.isOffsetInside(editor, marker.offsetR, cache) } }
+        }) {
+          tagVisitor?.scrollToClosest()
+        }
       }
     }
   }
 
   @ExternalUsage
   fun markResults(resultsToMark: SortedSet<Int>) {
-    tagger = Tagger(editor)
-    tagCanvas.setMarkers(emptyList())
-
-    val processor = SearchProcessor.fromRegex(editor, "", defaultBoundaries)
-      .apply { results = IntArrayList(resultsToMark) }
-
+    val jumpEditor = jumpEditors.singleOrNull() ?: return
+    markResults(mapOf(jumpEditor to resultsToMark))
+  }
+  
+  @ExternalUsage
```

Comment:
Can we add a test for this to [`ExternalUsageTest`](https://github.com/acejump/AceJump/blob/master/src/test/kotlin/ExternalUsageTest.kt)?

Reply:
I think I should fix all the existing tests first, turns out calling `splitters` on the IJ test editor throws an exception... will look into it after that.

Reply:
Do you know if the IJ test fixture can do multi-editor tests? I don't see an easy way to do that.

Reply:
You would need to create a new test class extending [`FileEditorManagerTestCase`](https://github.com/JetBrains/intellij-community/blob/master/platform/testFramework/src/com/intellij/testFramework/FileEditorManagerTestCase.java), then you can do multi-editor tests using [`openFiles(...)`](https://github.com/JetBrains/intellij-community/blob/37cc786e98fad74f18c1ba8042a4a017c46dc42b/platform/testFramework/src/com/intellij/testFramework/FileEditorManagerTestCase.java#L76).

Reply:
Added a test, let me know if it's ok.

Reply:
Looks good to me!